src/_pytest/python_api.py

```diff
@@ -100,6 +100,9 @@ def __eq__(self, actual) -> bool:
             a == self._approx_scalar(x) for a, x in self._yield_comparisons(actual)
         )
 
+    def __bool__(self):
+        raise AssertionError("use approx() like this: `assert a == approx(b)`")
```

Comment:
I think we can add a bit more context:

```suggestion
        raise AssertionError("approx() is not supported in a boolean context.\nUse approx() like this: `assert a == approx(b)`")
```

Reply:
I like that better

Reply:
Oh just realized, perhaps this should be `breaking`? It will break test suites which are misusing it.